### PR TITLE
[incubator/elasticsearch] Add updateStrategy for master and data statefulsets

### LINKED ISSUE #7148: [incubator/elasticsearch] Add updateStrategy to Stateful sets
FEATURE REQUEST

**Version of Helm and Kubernetes**:
Helm: 2.9.1
K8S: 1.10.6

**Which chart**:
incubator/elasticsearch

**What happened**:

By default Stateful sets created by this chart have updateStrategy: onDelete

**What you expected to happen**:

I'd like to configure updateStrategy via values.yaml

### PR DESCRIPTION
**Pull request by maver1ck**
**What this PR does / why we need it**:
Adding updateStrategy for statefulsets

**Which issue this PR fixes**: fixes #7148

### CODE REVIEW

File: incubator/elasticsearch/templates/master-statefulset.yaml [hunk 1]
```diff
@@ -141,6 +141,8 @@ spec:
       - name: data
         emptyDir: {}
   {{- else }}
```
Comment (davidkarlsen) [inline incubator/elasticsearch/templates/master-statefulset.yaml]: this depends on persistence being enabled - should it not be independent of that (even if it makes most sense with a "proper" setup with persistence?

Comment (maver1ck) [reply incubator/elasticsearch/templates/master-statefulset.yaml]: @davidkarlsen 
You're right. Fixed.

**Comment by davidkarlsen** [comment]
The test seems to have errors: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/helm_charts/7150/pull-charts-e2e/12528/?log#log - but so does also this other PR on the elastic chart: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/helm_charts/7085/pull-charts-e2e/12302/?log#log

It would be great if these could be fixed - wdyt?

**Comment by maver1ck** [comment]
I think it's impossible.
As far as I'm concerned there is no possibility to force helm to create masters before clients.

**Comment by maver1ck** [comment]
ping @davidkarlsen

**Comment by davidkarlsen** [comment]
@maver1ck can you fix the merge-conflict?

**Comment by maver1ck** [comment]
@davidkarlsen 
Done.

File: incubator/elasticsearch/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: elasticsearch
 home: https://www.elastic.co/products/elasticsearch
-version: 1.6.5
+version: 1.6.6
```
Comment (cpanato) [inline incubator/elasticsearch/Chart.yaml]: I would update to `1.7.0` since this is a new functionality

Comment (maver1ck) [reply incubator/elasticsearch/Chart.yaml]: Done

**Review by NicolasT — APPROVED** [review]
Very interested to see this land. Also, any reason to keep `OnDelete` the default instead of switching to `RollingUpdate`?

**Comment by maver1ck** [comment]
@NicolasT 
I left OnDelete to maintain reverse compatability. (it's default in K8S)

**Comment by rendhalver** [comment]
Nice work! Thanks @maver1ck!
/approved

**Pull request merged**